### PR TITLE
FIX: Layout overrides corrupting layout state.

### PR DESCRIPTION
**Pull request by Rene-Damm**
I got several questions at Unite around distinguishing devices based on how they are used (e.g. orientations) and added a sample that uses a layout override to distinguish a "Player1" gamepad from a "Player2" gamepad. However, doing so exposed multiple bugs with layout overrides.

So, here we go. New sample plus a number of fixes to layout overrides.

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/Editor/ControlPicker/InputControlPickerDropdown.cs [hunk 1]
```diff
@@ -197,6 +197,10 @@ private void AddDeviceTreeItemRecursive(InputControlLayout layout, AdvancedDropd
 
                     AddCharacterKeyBindingsTo(byCharacterGroup, keyboard);
                     AddPhysicalKeyBindingsTo(byLocationGroup, keyboard, searchable);
+
+                    // AnyKey won't appear in either group. Add it explicitly.
+                    AddControlItem(deviceItem, null,
+                        layout.FindControl(new InternedString("anyKey")).Value, layout.name, null, searchable);
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Editor/ControlPicker/InputControlPickerDropdown.cs:203]: Minor, unrelated fix. When I switched the control picker to a special path for keyboards, I didn't realize that now `anyKey` got culled from the list. This change adds the control back to the list.

**Review by jechter — APPROVED** [review]

**Review by douglas-piconi — APPROVED** [review]
